src/viam/app/app_client.py

```diff
@@ -374,11 +449,27 @@ async def list_organizations(self) -> List[Organization]:
         response: ListOrganizationsResponse = await self._app_client.ListOrganizations(request, metadata=self._metadata)
         return list(response.organizations)
 
+    # TODO(RSDK-5569): implement
     async def list_organizations_by_user(self, user_id: str) -> List[OrgDetails]:
         raise NotImplementedError()
 
-    async def get_organization(self) -> Organization:
-        raise NotImplementedError()
+    async def get_organization(self, org_id: Optional[str] = None) -> Organization:
+        """Return details about the requested organization.
+
+        Args:
+            org_id (Optional[str]): ID of the organization to query. If None, defaults to the
+            currently-authed org.
+
+        Raises:
+            GRPCError: If the provided org_id is invalid, or not currently authed to.
+
+        Returns:
+            viam.proto.app.Organization: The requested organization.
+        """
+        org_id = org_id if org_id else await self._get_organization_id()
```

Comment:
Similar issue as an above comment - is (`""`) a valid org_id? If so, we need to check for `not None` explicitly. If not, we might want to update the doc-string to something like `If None or blank/empty`.

```suggestion
        org_id = org_id if org_id is not None else await self._get_organization_id()
```